src/Adapter/Image/ImageRetriever.php

```diff
@@ -173,18 +179,20 @@ public function getImage($object, $id_image)
         $generateHighDpiImages = (bool) Configuration::get('PS_HIGHT_DPI');
 
         foreach ($image_types as $image_type) {
-            $resizedImagePath = implode(DIRECTORY_SEPARATOR, [
-                $imageFolderPath,
-                $id_image . '-' . $image_type['name'] . '.' . $ext,
-            ]);
+            if (!$psImageoptLazyResize) {
```

Comment:
So the behavior is "when `PS_IMAGEOPT_LAZY_RESIZE` is false we do not resize images", right?

Then maybe what we could do is
```
public function getImage($object, $id_image)
```
⬇️ 
```
public function getImage($object, $id_image, $performResizeAtRuntime)
```

And we use `if ($performResizeAtRuntime$` instead. The function that calls `getImage()` would fetch `PS_IMAGEOPT_LAZY_RESIZE`. This allows module that rely on `getImage()` to use or not the optimization feature.